vllm/pooling_params.py

```diff
@@ -1,15 +1,18 @@
 from typing import Any, Optional
 
+import msgspec
 
-class PoolingParams:
+
+class PoolingParams(
+        msgspec.Struct,
+        omit_defaults=True,  # type: ignore[call-arg]
+        array_like=True):  # type: ignore[call-arg]
     """Pooling parameters for pooling.
 
     Attributes:
         additional_data: Any additional data needed for pooling.
     """
-
-    def __init__(self, additional_data: Optional[Any] = None):
-        self.additional_data = additional_data
+    additional_data: Optional[Any] = None
```

Comment:
does msgpack work for this `Any` type data?

Reply:
Yeah I tried it, and looks like it works with any type that's working with msgspec. It fails with types that are not working with msgspec by default (e.g., array). but we can get around it by enc/dec hook